src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
> it's illegal for both to be XRP

Yeah, as I said in the prior comment, even though it doesn't happen, it can still be misleading if the logic doesn't follow what makes sense.

> Can't account for taste.

It's also fewer characters :)